File: src/main/java/com/finances/repository/ExpenseRepository.java
package com.finances.repository;

import com.finances.entity.Expense;
import org.springframework.data.jpa.repository.support.JpaRepositoryImplementation;
import org.springframework.stereotype.Repository;

import javax.transaction.Transactional;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

@Repository
@Transactional
public interface ExpenseRepository extends JpaRepositoryImplementation<Expense, Integer> {

    Optional<List<Expense>> findByExpireAtBetween(LocalDateTime startDate, LocalDateTime endDate);

}

File: src/main/kotlin/com/finances/repository/ExpenseRepository.kt
package com.finances.repository

import com.finances.entity.Expense
import org.springframework.data.jpa.repository.support.JpaRepositoryImplementation
import org.springframework.stereotype.Repository
import java.time.LocalDateTime
import javax.transaction.Transactional

@Repository
@Transactional
interface ExpenseRepository : JpaRepositoryImplementation<Expense, Int> {

    fun findByExpireAtBetween(startDate: LocalDateTime, endDate: LocalDateTime): List<Expense>
}
